### PR TITLE
bt-5: Create Select component

### LINKED ISSUE #5: Create Select component
Create a custom dropdown component which we will reuse in the project

User should be able to select one option or multiple options

Image

https://mui.com/material-ui/react-select/

### PR DESCRIPTION
**Pull request by MChornomaz**

### CODE REVIEW

**Comment by nikita-remeslov** [comment]
At least to approval you need for merge this PR

File: frontend/src/bundles/common/components/select/select.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    type FieldValues,
+    type Path,
+    type PathValue,
+} from 'react-hook-form';
+
+import { useFormController } from '~/bundles/common/hooks/hooks.js';
+
+import arrowIcon from '../../../../assets/img/select-arrow.svg';
+
+type SelectProperties<T extends FieldValues> = {
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/select/select.tsx]:
```suggestion
type Properties<T extends FieldValues> = {
```

File: frontend/src/bundles/common/components/select/select.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    control: Control<T, null>;
+    errors: FieldErrors<T>;
+    label: string;
+    name: FieldPath<T>;
+    options: { value: string | number; label: string }[];
+    multiple?: boolean;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/select/select.tsx]:
```suggestion
```

File: frontend/src/bundles/common/components/select/select.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+const SelectContainer = styled(FormControl)(({ theme }) => ({
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/select/select.tsx]: we are using style modules

File: frontend/src/bundles/common/components/select/select.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    fontStyle: 'normal',
+    lineHeight: 'normal',
+    top: '-5px',
+    color: '#1f1e29',
+    fontWeight: 500,
+});
+
+const shadowOpacity = 0.25;
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/select/select.tsx]:
```suggestion
```

File: frontend/src/bundles/common/components/select/select.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        control,
+        defaultValue: (multiple
+            ? [options[firstElementIndex].value]
+            : options[firstElementIndex].value) as PathValue<T, Path<T>>,
+    });
+
+    const error = errors[name]?.message;
+    const hasError = Boolean(error);
+
+    return (
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/select/select.tsx:48]: we just need to add select. Label will be added in another ticket

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by JosuerBague — APPROVED** [review]

**Comment by MChornomaz** [comment]
Nikita, could you please check the changes?

**Comment by nikita-remeslov** [comment]
@MChornomaz Looks like status for your issue In progress

**Comment by MChornomaz** [comment]
@nikita-remeslov  should I move it to In Review?

**Comment by nikita-remeslov** [comment]
@MChornomaz  now ticket In Review. I will review this ticket

**Comment by MChornomaz** [comment]
Thanks, I moved it there after the first day, when I did it with styled from MI, and didn't notice that it was moved back

**Comment by nikita-remeslov** [comment]
I moved back when I requested changes to your code

**Comment by MChornomaz** [comment]
Thank you for the information. I will keep that in mind and will not make this mistake again)

File: frontend/src/bundles/common/components/select/select.module.scss [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    transform: rotate(0deg);
+    transition: transform 0.3s ease;
+}
+
+.container div[aria-expanded="true"] ~ .icon {
+    transform: rotate(180deg);
+}
+
+.container fieldset {
+    border: 1px solid #d5dce8;
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/select/select.module.scss]: it is our common color

File: frontend/src/bundles/common/components/select/select.tsx (hunk 5 above)
Comment (nikita-remeslov) [reply frontend/src/bundles/common/components/select/select.tsx:48]: Not resolved

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/select/select.tsx (hunk 5 above)
Comment (MChornomaz) [reply frontend/src/bundles/common/components/select/select.tsx:48]: @nikita-remeslov 
image
Should it be in this way?

Comment (nikita-remeslov) [reply frontend/src/bundles/common/components/select/select.tsx:48]: Looks like you can use out FormControl and FormLabel - https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/blob/development/frontend/src/bundles/common/components/form-control/form-control.tsx

**Comment by vladjsp** [comment]
It's all about stylings. But am I seeing correct that on left side of select we have 2 paddings and text is far from left side? In Figma it's like 14px but in component you have 28px (14px for parent div and 14px for element).

**Comment by MChornomaz** [comment]
Thanks, I will check it)

File: frontend/src/bundles/common/components/select/select.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    type Control,
+    type FieldPath,
+    type FieldValues,
+    type Path,
+    type PathValue,
+} from 'react-hook-form';
+
+import { useFormController } from '~/bundles/common/hooks/hooks.js';
+
+import arrowIcon from '../../../../assets/img/select-arrow.svg';
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/select/select.tsx]: use alias to import `~`

**Review by nikita-remeslov — COMMENTED** [review]
Merge develop and update the styles with new variables

**Comment by MChornomaz** [comment]
@nikita-remeslov done

File: frontend/src/bundles/common/components/select/select.module.scss [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    font-size: var(--font-size-base);
+    font-family: var(--font-family);
+}
+
+.container .input {
+    height: 38px;
+    padding: 10px 7px 12px 14px;
+    color: var(--nepal, #90a3bf);
+    font-weight: var(--font-weight-regular);
+    border: 1px solid var(--botticelli, #d5dce8);
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/select/select.module.scss]:
```suggestion
    border: 1px solid var(--botticelli);
```

File: frontend/src/bundles/common/components/select/select.module.scss [hunk 9]
```diff
@@ -0,0 +1,41 @@
+.container {
+    width: 100%;
+    font-size: var(--font-size-base);
+    font-family: var(--font-family);
+}
+
+.container .input {
+    height: 38px;
+    padding: 10px 7px 12px 14px;
+    color: var(--nepal, #90a3bf);
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/select/select.module.scss]:
```suggestion
    color: var(--nepal);
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**